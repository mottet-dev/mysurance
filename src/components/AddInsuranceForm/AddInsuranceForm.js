import React, { Component } from 'react';
import { View, Picker } from 'react-native';
import { FormInput, FormLabel, Button } from 'react-native-elements';

import { textColor, strongColor, lightColor } from '../../constants';

class AddInsuranceForm extends Component {
    render() {
        const { 
            labelStyle, 
            buttonStyle, 
            buttonTextStyle, 
            formStyle,
            formInputContainerStyle,
            formInputStyle,
            pickerStyle,
            pickerItemStyle } = styles;

        return (
            <View style={formStyle}>
                <View>
                    <FormLabel labelStyle={labelStyle}>Title:</FormLabel>
                    <FormInput
                        containerStyle={formInputContainerStyle}
                        inputStyle={formInputStyle}
                        underlineColorAndroid={'transparent'}
                        keyboardType="default"
                    />

                    <FormLabel labelStyle={labelStyle}>Premium (yearly):</FormLabel>
                    <FormInput
                        containerStyle={formInputContainerStyle}
                        inputStyle={formInputStyle}
                        underlineColorAndroid={'transparent'}
                        keyboardType="numeric"
                    />

                    <FormLabel labelStyle={labelStyle}>Category:</FormLabel>
                    <Picker
                        style={pickerStyle}
                        pickerItemStyle={pickerItemStyle}
                        mode="dropdown"
                    >
                        <Picker.Item label="Health Insurance" value="healthInsurance" />
                        <Picker.Item label="Car Insurance" value="carInsurance" />
                        <Picker.Item label="House Insurance" value="houseInsurance" />
                    </Picker>
                </View>

                <View>
                    <Button 
                        title="Done"
                        buttonStyle={buttonStyle}
                        textStyle={buttonTextStyle}
                    />
                </View>
            </View>
        );
    }
}

const styles = {
    labelStyle: {
        color: textColor,
        fontSize: 18
    },
    buttonStyle: {
        backgroundColor: strongColor,
        height: 60
    },
    buttonTextStyle: {
        fontSize: 18
    },
    formStyle: {
        flex: 1,
        flexDirection: 'column',
        justifyContent: 'space-between',
        marginBottom: 20
    },
    formInputContainerStyle: {
        backgroundColor: lightColor,
        borderWidth: 0.5,
        borderColor: '#95989A'
    },
    formInputStyle: {
        color: 'black',
        fontSize: 16,
    },
    pickerStyle: {
        backgroundColor: lightColor,
        marginRight: 15,
        marginLeft: 15,
        borderWidth: 0.5,
        borderColor: '#95989A',
    },
    pickerItemStyle: {
        color: 'black',
        fontSize: 16,
        borderWidth: 0.5,
    }
}

export default AddInsuranceForm;