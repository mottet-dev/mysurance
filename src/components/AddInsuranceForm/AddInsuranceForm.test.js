import React from 'react';
import { shallow } from 'enzyme';
import AddInsuranceForm from './AddInsuranceForm';
import { FormInput, FormLabel, Button } from 'react-native-elements';

describe('<AddInsuranceForm />', () => {
    const component = shallow(<AddInsuranceForm />);

    it('renders properly', () => {
        expect(component).toMatchSnapshot();
    });

    it('contains 2 <FormInput />', () => {
        expect(component.find('View > View').children(FormInput).length).toEqual(2);
    });

    it('contains 3 <FormLabel />', () => {
        expect(component.find('View > View').children(FormLabel).length).toEqual(3);
    });

    it('contains 1 <Picker />', () => {
        expect(component.find('View > View').children('Picker').length).toEqual(1);
    });

    it('contains 1 <Button />', () => {
        expect(component.find('View').children(Button).length).toEqual(1);
    });
});
