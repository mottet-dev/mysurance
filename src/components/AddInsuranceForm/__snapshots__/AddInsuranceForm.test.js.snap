// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddInsuranceForm /> renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddInsuranceForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <View>
          <FormLabel
            labelStyle={
              Object {
                "color": "#EE712B",
                "fontSize": 18,
              }
            }
          >
            Title:
          </FormLabel>
          <FormInput
            containerStyle={
              Object {
                "backgroundColor": "#FBF7F7",
                "borderColor": "#95989A",
                "borderWidth": 0.5,
              }
            }
            inputStyle={
              Object {
                "color": "black",
                "fontSize": 16,
              }
            }
            keyboardType="default"
            label="title"
            normalizeFontSize={true}
            onChangeText={[Function]}
            underlineColorAndroid="transparent"
            value=""
          />
          <FormLabel
            labelStyle={
              Object {
                "color": "#EE712B",
                "fontSize": 18,
              }
            }
          >
            Premium (yearly):
          </FormLabel>
          <FormInput
            containerStyle={
              Object {
                "backgroundColor": "#FBF7F7",
                "borderColor": "#95989A",
                "borderWidth": 0.5,
              }
            }
            inputStyle={
              Object {
                "color": "black",
                "fontSize": 16,
              }
            }
            keyboardType="numeric"
            label="premium"
            normalizeFontSize={true}
            onChangeText={[Function]}
            underlineColorAndroid="transparent"
            value=""
          />
          <FormLabel
            labelStyle={
              Object {
                "color": "#EE712B",
                "fontSize": 18,
              }
            }
          >
            Category:
          </FormLabel>
          <Picker
            mode="dropdown"
            onValueChange={[Function]}
            pickerItemStyle={
              Object {
                "color": "black",
                "fontSize": 16,
              }
            }
            selectedValue="healthinsurance"
            style={
              Object {
                "backgroundColor": "#FBF7F7",
                "borderColor": "#95989A",
                "borderWidth": 0.5,
                "marginLeft": 15,
                "marginRight": 15,
              }
            }
          >
            &gt;
            <PickerItem
              label="Health Insurance"
              value="healthinsurance"
            />
            <PickerItem
              label="Car Insurance"
              value="carinsurance"
            />
            <PickerItem
              label="House Insurance"
              value="houseinsurance"
            />
          </Picker>
        </View>,
        <View>
          <Button
            buttonStyle={
              Object {
                "backgroundColor": "#DC4712",
                "height": 60,
              }
            }
            textStyle={
              Object {
                "fontSize": 18,
              }
            }
            title="Done"
          />
        </View>,
      ],
      "style": Object {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-between",
        "marginBottom": 20,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <FormLabel
              labelStyle={
                Object {
                  "color": "#EE712B",
                  "fontSize": 18,
                }
              }
            >
              Title:
            </FormLabel>,
            <FormInput
              containerStyle={
                Object {
                  "backgroundColor": "#FBF7F7",
                  "borderColor": "#95989A",
                  "borderWidth": 0.5,
                }
              }
              inputStyle={
                Object {
                  "color": "black",
                  "fontSize": 16,
                }
              }
              keyboardType="default"
              label="title"
              normalizeFontSize={true}
              onChangeText={[Function]}
              underlineColorAndroid="transparent"
              value=""
            />,
            <FormLabel
              labelStyle={
                Object {
                  "color": "#EE712B",
                  "fontSize": 18,
                }
              }
            >
              Premium (yearly):
            </FormLabel>,
            <FormInput
              containerStyle={
                Object {
                  "backgroundColor": "#FBF7F7",
                  "borderColor": "#95989A",
                  "borderWidth": 0.5,
                }
              }
              inputStyle={
                Object {
                  "color": "black",
                  "fontSize": 16,
                }
              }
              keyboardType="numeric"
              label="premium"
              normalizeFontSize={true}
              onChangeText={[Function]}
              underlineColorAndroid="transparent"
              value=""
            />,
            <FormLabel
              labelStyle={
                Object {
                  "color": "#EE712B",
                  "fontSize": 18,
                }
              }
            >
              Category:
            </FormLabel>,
            <Picker
              mode="dropdown"
              onValueChange={[Function]}
              pickerItemStyle={
                Object {
                  "color": "black",
                  "fontSize": 16,
                }
              }
              selectedValue="healthinsurance"
              style={
                Object {
                  "backgroundColor": "#FBF7F7",
                  "borderColor": "#95989A",
                  "borderWidth": 0.5,
                  "marginLeft": 15,
                  "marginRight": 15,
                }
              }
            >
              &gt;
              <PickerItem
                label="Health Insurance"
                value="healthinsurance"
              />
              <PickerItem
                label="Car Insurance"
                value="carinsurance"
              />
              <PickerItem
                label="House Insurance"
                value="houseinsurance"
              />
            </Picker>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Title:",
              "labelStyle": Object {
                "color": "#EE712B",
                "fontSize": 18,
              },
            },
            "ref": null,
            "rendered": "Title:",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "containerStyle": Object {
                "backgroundColor": "#FBF7F7",
                "borderColor": "#95989A",
                "borderWidth": 0.5,
              },
              "inputStyle": Object {
                "color": "black",
                "fontSize": 16,
              },
              "keyboardType": "default",
              "label": "title",
              "normalizeFontSize": true,
              "onChangeText": [Function],
              "underlineColorAndroid": "transparent",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Premium (yearly):",
              "labelStyle": Object {
                "color": "#EE712B",
                "fontSize": 18,
              },
            },
            "ref": null,
            "rendered": "Premium (yearly):",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "containerStyle": Object {
                "backgroundColor": "#FBF7F7",
                "borderColor": "#95989A",
                "borderWidth": 0.5,
              },
              "inputStyle": Object {
                "color": "black",
                "fontSize": 16,
              },
              "keyboardType": "numeric",
              "label": "premium",
              "normalizeFontSize": true,
              "onChangeText": [Function],
              "underlineColorAndroid": "transparent",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Category:",
              "labelStyle": Object {
                "color": "#EE712B",
                "fontSize": 18,
              },
            },
            "ref": null,
            "rendered": "Category:",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                ">",
                <PickerItem
                  label="Health Insurance"
                  value="healthinsurance"
                />,
                <PickerItem
                  label="Car Insurance"
                  value="carinsurance"
                />,
                <PickerItem
                  label="House Insurance"
                  value="houseinsurance"
                />,
              ],
              "mode": "dropdown",
              "onValueChange": [Function],
              "pickerItemStyle": Object {
                "color": "black",
                "fontSize": 16,
              },
              "selectedValue": "healthinsurance",
              "style": Object {
                "backgroundColor": "#FBF7F7",
                "borderColor": "#95989A",
                "borderWidth": 0.5,
                "marginLeft": 15,
                "marginRight": 15,
              },
            },
            "ref": null,
            "rendered": Array [
              ">",
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "class",
                "props": Object {
                  "label": "Health Insurance",
                  "value": "healthinsurance",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "2",
                "nodeType": "class",
                "props": Object {
                  "label": "Car Insurance",
                  "value": "carinsurance",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "3",
                "nodeType": "class",
                "props": Object {
                  "label": "House Insurance",
                  "value": "houseinsurance",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Button
            buttonStyle={
              Object {
                "backgroundColor": "#DC4712",
                "height": 60,
              }
            }
            textStyle={
              Object {
                "fontSize": 18,
              }
            }
            title="Done"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "buttonStyle": Object {
              "backgroundColor": "#DC4712",
              "height": 60,
            },
            "textStyle": Object {
              "fontSize": 18,
            },
            "title": "Done",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <View>
            <FormLabel
              labelStyle={
                Object {
                  "color": "#EE712B",
                  "fontSize": 18,
                }
              }
            >
              Title:
            </FormLabel>
            <FormInput
              containerStyle={
                Object {
                  "backgroundColor": "#FBF7F7",
                  "borderColor": "#95989A",
                  "borderWidth": 0.5,
                }
              }
              inputStyle={
                Object {
                  "color": "black",
                  "fontSize": 16,
                }
              }
              keyboardType="default"
              label="title"
              normalizeFontSize={true}
              onChangeText={[Function]}
              underlineColorAndroid="transparent"
              value=""
            />
            <FormLabel
              labelStyle={
                Object {
                  "color": "#EE712B",
                  "fontSize": 18,
                }
              }
            >
              Premium (yearly):
            </FormLabel>
            <FormInput
              containerStyle={
                Object {
                  "backgroundColor": "#FBF7F7",
                  "borderColor": "#95989A",
                  "borderWidth": 0.5,
                }
              }
              inputStyle={
                Object {
                  "color": "black",
                  "fontSize": 16,
                }
              }
              keyboardType="numeric"
              label="premium"
              normalizeFontSize={true}
              onChangeText={[Function]}
              underlineColorAndroid="transparent"
              value=""
            />
            <FormLabel
              labelStyle={
                Object {
                  "color": "#EE712B",
                  "fontSize": 18,
                }
              }
            >
              Category:
            </FormLabel>
            <Picker
              mode="dropdown"
              onValueChange={[Function]}
              pickerItemStyle={
                Object {
                  "color": "black",
                  "fontSize": 16,
                }
              }
              selectedValue="healthinsurance"
              style={
                Object {
                  "backgroundColor": "#FBF7F7",
                  "borderColor": "#95989A",
                  "borderWidth": 0.5,
                  "marginLeft": 15,
                  "marginRight": 15,
                }
              }
            >
              &gt;
              <PickerItem
                label="Health Insurance"
                value="healthinsurance"
              />
              <PickerItem
                label="Car Insurance"
                value="carinsurance"
              />
              <PickerItem
                label="House Insurance"
                value="houseinsurance"
              />
            </Picker>
          </View>,
          <View>
            <Button
              buttonStyle={
                Object {
                  "backgroundColor": "#DC4712",
                  "height": 60,
                }
              }
              textStyle={
                Object {
                  "fontSize": 18,
                }
              }
              title="Done"
            />
          </View>,
        ],
        "style": Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "space-between",
          "marginBottom": 20,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <FormLabel
                labelStyle={
                  Object {
                    "color": "#EE712B",
                    "fontSize": 18,
                  }
                }
              >
                Title:
              </FormLabel>,
              <FormInput
                containerStyle={
                  Object {
                    "backgroundColor": "#FBF7F7",
                    "borderColor": "#95989A",
                    "borderWidth": 0.5,
                  }
                }
                inputStyle={
                  Object {
                    "color": "black",
                    "fontSize": 16,
                  }
                }
                keyboardType="default"
                label="title"
                normalizeFontSize={true}
                onChangeText={[Function]}
                underlineColorAndroid="transparent"
                value=""
              />,
              <FormLabel
                labelStyle={
                  Object {
                    "color": "#EE712B",
                    "fontSize": 18,
                  }
                }
              >
                Premium (yearly):
              </FormLabel>,
              <FormInput
                containerStyle={
                  Object {
                    "backgroundColor": "#FBF7F7",
                    "borderColor": "#95989A",
                    "borderWidth": 0.5,
                  }
                }
                inputStyle={
                  Object {
                    "color": "black",
                    "fontSize": 16,
                  }
                }
                keyboardType="numeric"
                label="premium"
                normalizeFontSize={true}
                onChangeText={[Function]}
                underlineColorAndroid="transparent"
                value=""
              />,
              <FormLabel
                labelStyle={
                  Object {
                    "color": "#EE712B",
                    "fontSize": 18,
                  }
                }
              >
                Category:
              </FormLabel>,
              <Picker
                mode="dropdown"
                onValueChange={[Function]}
                pickerItemStyle={
                  Object {
                    "color": "black",
                    "fontSize": 16,
                  }
                }
                selectedValue="healthinsurance"
                style={
                  Object {
                    "backgroundColor": "#FBF7F7",
                    "borderColor": "#95989A",
                    "borderWidth": 0.5,
                    "marginLeft": 15,
                    "marginRight": 15,
                  }
                }
              >
                &gt;
                <PickerItem
                  label="Health Insurance"
                  value="healthinsurance"
                />
                <PickerItem
                  label="Car Insurance"
                  value="carinsurance"
                />
                <PickerItem
                  label="House Insurance"
                  value="houseinsurance"
                />
              </Picker>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Title:",
                "labelStyle": Object {
                  "color": "#EE712B",
                  "fontSize": 18,
                },
              },
              "ref": null,
              "rendered": "Title:",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "containerStyle": Object {
                  "backgroundColor": "#FBF7F7",
                  "borderColor": "#95989A",
                  "borderWidth": 0.5,
                },
                "inputStyle": Object {
                  "color": "black",
                  "fontSize": 16,
                },
                "keyboardType": "default",
                "label": "title",
                "normalizeFontSize": true,
                "onChangeText": [Function],
                "underlineColorAndroid": "transparent",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Premium (yearly):",
                "labelStyle": Object {
                  "color": "#EE712B",
                  "fontSize": 18,
                },
              },
              "ref": null,
              "rendered": "Premium (yearly):",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "containerStyle": Object {
                  "backgroundColor": "#FBF7F7",
                  "borderColor": "#95989A",
                  "borderWidth": 0.5,
                },
                "inputStyle": Object {
                  "color": "black",
                  "fontSize": 16,
                },
                "keyboardType": "numeric",
                "label": "premium",
                "normalizeFontSize": true,
                "onChangeText": [Function],
                "underlineColorAndroid": "transparent",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Category:",
                "labelStyle": Object {
                  "color": "#EE712B",
                  "fontSize": 18,
                },
              },
              "ref": null,
              "rendered": "Category:",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  ">",
                  <PickerItem
                    label="Health Insurance"
                    value="healthinsurance"
                  />,
                  <PickerItem
                    label="Car Insurance"
                    value="carinsurance"
                  />,
                  <PickerItem
                    label="House Insurance"
                    value="houseinsurance"
                  />,
                ],
                "mode": "dropdown",
                "onValueChange": [Function],
                "pickerItemStyle": Object {
                  "color": "black",
                  "fontSize": 16,
                },
                "selectedValue": "healthinsurance",
                "style": Object {
                  "backgroundColor": "#FBF7F7",
                  "borderColor": "#95989A",
                  "borderWidth": 0.5,
                  "marginLeft": 15,
                  "marginRight": 15,
                },
              },
              "ref": null,
              "rendered": Array [
                ">",
                Object {
                  "instance": null,
                  "key": "1",
                  "nodeType": "class",
                  "props": Object {
                    "label": "Health Insurance",
                    "value": "healthinsurance",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "2",
                  "nodeType": "class",
                  "props": Object {
                    "label": "Car Insurance",
                    "value": "carinsurance",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "3",
                  "nodeType": "class",
                  "props": Object {
                    "label": "House Insurance",
                    "value": "houseinsurance",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Button
              buttonStyle={
                Object {
                  "backgroundColor": "#DC4712",
                  "height": 60,
                }
              }
              textStyle={
                Object {
                  "fontSize": 18,
                }
              }
              title="Done"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "buttonStyle": Object {
                "backgroundColor": "#DC4712",
                "height": 60,
              },
              "textStyle": Object {
                "fontSize": 18,
              },
              "title": "Done",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
